examples/shaders/sky-shader/src/lib.rs

```diff
@@ -160,15 +160,11 @@ pub fn fs(screen_pos: Vec2) -> Vec4 {
         Vec4::new(0.0, -0.14834046, -0.98893654, 0.0),
     );
 
-    let clip_pos = screen_pos.extend(1.0).extend(1.0);
-    let world_pos = {
-        let p = clip_to_world.mul_vec4(clip_pos);
-        p.truncate() / p.w()
-    };
+    let world_pos = clip_to_world.transform_point3(screen_pos.extend(1.0));
     let dir = (world_pos - eye_pos).normalize();
 
     // evaluate Preetham sky model
-    let color = sky(dir, sun_pos);
+    let color = sky(dir.into(), sun_pos);
```

Comment:
I don't seem to need this `.into()` here, `dir` is a `Vec3` which `sky` expects.

Completely unrelated note: I tried using the new `let eye_pos = const_vec3!([0.0, 0.0997, 0.2]);` macro here which works fine on the host but only in SPIR-V with `const VARIABLE: Vec3 = const_vec3!([...]);`:

```rust
  error: Cannot cast between pointer types
     --> examples/shaders/sky-shader/src/lib.rs:154:19
      |
  154 |     let eye_pos = const_vec3!([0.0, 0.0997, 0.2]);
      |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |
      = note: from: *{Function} [u8; 16]
      = note: to: *{Function} [f32; 3]
```

That's a deliberate error from `zombie_bitcast_ptr`: I don't fully grasp what's going on, is this something you want documented in an issue to pursue later on?

Reply:
File an issue, at the very least maybe the error message can be improved.